### DISCUSSION TITLE
Syntax-highlight for LICENSE.*

### DISCUSSION THREAD

**Discussion by shynur** [community/community#62161] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

By default, GitHub will highlight some keywords for `LICENSE`,
e.g., \<<https://github.com/MatthewZMD/.github/blob/main/LICENSE>\>.

But I found that if I add a suffix `.txt`, the file won't be highlit,
e.g., \<<https://github.com/shynur/.emacs.d/blob/main/LICENSE.txt>\>.

Can we add the support to highlight for `LICENSE.txt`?

**Comment by priyankarpal** [comment]
> ### Select Topic Area
> 
> Product Feedback
> ### Body
> 
> By default, GitHub will highlight some keywords for `LICENSE`, e.g., <https://github.com/MatthewZMD/.github/blob/main/LICENSE>.
> 
> But I found that if I add a suffix `.txt`, the file won't be highlit, e.g., <https://github.com/shynur/.emacs.d/blob/main/LICENSE.txt>.
> 
> Can we add the support to highlight for `LICENSE.txt`?

you don't need to add `.txt` for `LICENSE`. By default, GitHub will not apply syntax highlighting to plain text files like `.txt`.

Hope it will help you

**Comment by shynur** [comment]
> you don't need to add `.txt` for `LICENSE`

I need.

It is easier to choose a default editor to open file which ends with a suffix.

**Reply by priyankarpal** [reply]
By default, GitHub will not apply syntax highlighting to plain text files like .txt.

**Reply by shynur** [reply]
> Hope it will help you

Yes.  thx.

> By default, GitHub will not apply syntax highlighting to plain text files like .txt.

So, this is a feature request.
Are you a GitHub employee?  I wonder if we can implement this feature.
(`LICENSE.txt` only)

**Reply by priyankarpal** [reply]
The license file should be named "LICENSE" (for plain text) without any additional file extensions. GitHub and other tools specifically look for these filenames to interpret the license correctly.

Therefore, it is unnecessary to add the `.txt` extension to the license file, and doing so will not provide any additional benefit for the licensing purposes on GitHub. It's best to stick with the standard naming conventions, such as "LICENSE" for plain text.

**Reply by shynur** [reply]
> The license file should be named "LICENSE" (for plain text) without any additional file extensions. GitHub and other tools specifically look for these filenames to interpret the license correctly.

Are u sure?  I've read GitHub doc a lot and it sometimes recommend we do that:

See https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository and https://choosealicense.com/licenses/agpl-3.0/ :

> In the file name field, type LICENSE or LICENSE.md (with all caps).

> Create a text file (typically named LICENSE or LICENSE.txt) in the root of your source code and copy the text of the license into the file.

___

> it is unnecessary to add the .txt extension to the license file

Unnecessary to some people, but not me.
I've said:

> It is easier to choose a default editor to open file which ends with a suffix.

On MS-Windows, I just need to double-click a `*.txt` to open it, but what if the opposite??

___

> doing so will not provide any additional benefit for the licensing purposes on GitHub. It's best to stick with the standard naming conventions, such as "LICENSE" for plain text.

Why is it best?  Stripping suffix will provide any additional benefit for the licensing purposes?

**Reply by priyankarpal** [reply]
you can add LICENSE.md too

**Reply by priyankarpal** [reply]
@shynur Using standard naming conventions like "LICENSE" makes it easy for others to recognize and find the license file. Stripping suffixes won't provide any additional benefit for licensing purposes on GitHub or most version control systems.

**Reply by shynur** [reply]
> you can add LICENSE.md too

GitHub doesn't provide a license file that comply with markdown syntax.  The resulting rendered page is not pleasing to eyes.

**Reply by shynur** [reply]
alas... Just tell me if it's possible for GitHub to implement

> highlight for `LICENSE.txt`

If NO, I will mark it as answer and give up, thus the discussion ends.

**Reply by priyankarpal** [reply, marked as answer]
GitHub did not provide a built-in feature for highlighting the contents of a `LICENSE.txt` file directly within their web interface. This is because `LICENSE.txt` files are typically plain text files that do not have syntax highlighting like code files (e.g., `.js`, `.py`, `.java`, etc.).

However, keep in mind that GitHub continuously updates and improves its platform, so it's possible that new features have been introduced since my last update. To get the most accurate and up-to-date information, I recommend checking the GitHub documentation or their official website for any recent changes or feature updates.
